src/components/Form/FormRatio.react.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import * as React from "react";
+import cn from "classnames";
+import { Grid } from "../";
+
+type Props = {|
+  +className?: string,
+  +step?: number,
+  +min?: number,
+  +max?: number,
+  +defaultValue?: number,
+  +value?: number,
+  +onChange?: (SyntheticInputEvent<EventTarget>) => mixed,
+|};
+
+type State = {|
+  internalValue: number,
+  hasChanged: boolean,
+|};
+
+class FormRatio extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      internalValue: props.defaultValue || props.value || 0,
+      hasChanged: false,
+    };
+  }
+  handleOnChange = (e: SyntheticInputEvent<EventTarget>) => {
+    const { onChange = false } = this.props;
+    const value: number = Number(e.target.value);
+    if (onChange) {
+      onChange(e);
+      this.setState({ hasChanged: true });
+    } else {
+      this.setState({ internalValue: value, hasChanged: true });
+    }
+  };
```

Comment:
[nit] insert empty line after this

Reply:
This appears to be something prettier is doing, shall investigate an option to change it